kornia/filters/filter.py

```diff
@@ -122,7 +122,8 @@ def filter2d(
     input = input.view(-1, tmp_kernel.size(0), input.size(-2), input.size(-1))
 
     # convolve the tensor with the kernel.
-    output = F.conv2d(input, tmp_kernel, groups=tmp_kernel.size(0), padding=0, stride=1)
+    # NOTE: type(...) to fix getting `torch.bfloat16` type.
+    output = F.conv2d(input, tmp_kernel, groups=tmp_kernel.size(0), padding=0, stride=1).type(input.dtype)
```

Comment:
I think this isn't the right call here, this one just fail because it's the case [covered on the test,](https://github.com/kornia/kornia/blob/d4d272006549624e603ec6870bf8aa68cc576594/test/augmentation/test_base.py#L86) but probably others augmentations will fail as well -- instead of this, I think we can just skip the autocast test for now and work on that in a further PR -- i started to play around with this [here](https://github.com/shijianjian/kornia/compare/feat/aug_base_refactor...johnnv1:kornia:fix-autocast), and I think will need to add tests and fix for all base augmentation classes

Reply:
Okay. I will revert this one and merge. Leaving this issue to you then. XD.